### ISSUE TITLE
State properties required for PAKES used

### ISSUE THREAD

**Issue by menonsamir** [tlswg/tls-pake#4]
From IETF discussion with @BjoernMHaase and @FrankXL:

We should state explicitly what properties a PAKE must have to be eligible for use under this extension (and registered as a NamedPAKE).

The main one we've discussed so far is that the PAKE should be compatible with performing key confirmation through the standard TLS 1.3 Finished message.

**Comment by BjoernMHaase** [comment]
Yes I agree. I don't see any property except for the fact that the last PAKE protocol message needs to only include a key confirmation.

We also should be having a security consideration paragraph per protocol with an analysis on how the security analysis of the PAKE protocol is covering the security analysis of the interleaved TLS and PAKE construction.

**Comment by github-147** [comment]
As discussed above, a PAKE should be compatible with performing key confirmation through the standard TLS 1.3 Finished message, so do the key confirmation messages of the PAKE not need to compute individually. 
For example, the TLS 1.3 key schedule can be modified with the intermediate session key ISK for CPace and the shared symmetric secret K_main for SPAKE2+, respectively, and the key confirmation of these PAKEs can be achieved by the TLS 1.3 Finished message without computing the PAKEs’ key confirmation messages individually.

**Comment by BjoernMHaase** [comment]
Hi Chris, Yes I agree with you. The aspect that I wanted to highlight is that for any three-message flow PAKE such as SPAKE2+ or OPAQUE you have something "interleaved" where you have two protocols. What I meant is that for the specific PAKE instance with three flows some discussion will become important for demonstrating that the way that this interleaving combination is implemented is not security-relevant. IMO this will boil down to a discussion that the exact way how the explictit key confirmation is carried out is not relevant for the PAKE protocol proof and that the specific construction of the TLS "finished" message fullfils the requirements needed for the PAKE protocol's security proof.

**Comment by FrankXL** [comment]
> Hi Chris, Yes I agree with you. The aspect that I wanted to highlight is that for any three-message flow PAKE such as SPAKE2+ or OPAQUE you have something "interleaved" where you have two protocols. What I meant is that for the specific PAKE instance with three flows some discussion will become important for demonstrating that the way that this interleaving combination is implemented is not security-relevant. IMO this will boil down to a discussion that the exact way how the explictit key confirmation is carried out is not relevant for the PAKE protocol proof and that the specific construction of the TLS "finished" message fullfils the requirements needed for the PAKE protocol's security proof.

fully agree with this point. But do you think we need more consideration and discussion on other potential interleving combination?

**Comment by chris-wood** [comment]
> But do you think we need more consideration and discussion on other potential interleving combination?

Why would we consider alternative interleaving mechanisms? Is the current mechanism in some way inadequate?

**Comment by chris-wood** [comment]
> Hi Chris, Yes I agree with you. The aspect that I wanted to highlight is that for any three-message flow PAKE such as SPAKE2+ or OPAQUE you have something "interleaved" where you have two protocols. What I meant is that for the specific PAKE instance with three flows some discussion will become important for demonstrating that the way that this interleaving combination is implemented is not security-relevant. IMO this will boil down to a discussion that the exact way how the explictit key confirmation is carried out is not relevant for the PAKE protocol proof and that the specific construction of the TLS "finished" message fullfils the requirements needed for the PAKE protocol's security proof.

I don't disagree, but you seem to be simply advocating for some text in the security considerations which argues (or points to external arguments) for why replacing the PAKE-specific key confirmation with TLS's key confirmation is sufficient. Is that a correct summary?
